From 2cf91c140887395237f20673aaf315bccaaf1fec Mon Sep 17 00:00:00 2001
From: ctang <ctang@cloudera.com>
Date: Mon, 31 Oct 2016 17:03:02 -0400
Subject: [PATCH 692/723] CDH-46449: HIVE-15061: Metastore types are sometimes
 case sensitive (Chaoyu Tang, review by Mohit
 Sabharwal)

Change-Id: Idf4861ecdf47e10986dfebc8e84dd9df46edfedb
(cherry picked from commit 38dede72edbbeb93cb02a0e858b33778f562f61a)
---
 .../hadoop/hive/metastore/HiveAlterHandler.java    |    6 +++---
 .../hadoop/hive/metastore/HiveMetaStore.java       |    2 ++
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java
index dc82513..c7d2a50 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java
@@ -581,8 +581,8 @@ private void updatePartColumnStatsForAlterColumns(RawStore msdb, Partition oldPa
         for (ColumnStatisticsObj statsObj : statsObjs) {
           boolean found =false;
           for (FieldSchema newCol : newCols) {
-            if (statsObj.getColName().equals(newCol.getName())
-                && statsObj.getColType().equals(newCol.getType())) {
+            if (statsObj.getColName().equalsIgnoreCase(newCol.getName())
+                && statsObj.getColType().equalsIgnoreCase(newCol.getType())) {
               found = true;
               break;
             }
@@ -679,7 +679,7 @@ private void alterTableUpdateTableColumnStats(RawStore msdb,
                 boolean found = false;
                 for (FieldSchema newCol : newCols) {
                   if (statsObj.getColName().equalsIgnoreCase(newCol.getName())
-                      && statsObj.getColType().equals(newCol.getType())) {
+                      && statsObj.getColType().equalsIgnoreCase(newCol.getType())) {
                     found = true;
                     break;
                   }
diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
index 65a7555..b77db57 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
@@ -4484,6 +4484,7 @@ public boolean update_table_column_statistics(ColumnStatistics colStats)
       for (ColumnStatisticsObj statsObj:statsObjs) {
         colName = statsObj.getColName().toLowerCase();
         statsObj.setColName(colName);
+        statsObj.setColType(statsObj.getColType().toLowerCase());
         startFunction("write_column_statistics:  db=" + dbName + " table=" + tableName +
           " column=" + colName);
       }
@@ -4529,6 +4530,7 @@ public boolean update_partition_column_statistics(ColumnStatistics colStats)
       for (ColumnStatisticsObj statsObj:statsObjs) {
         colName = statsObj.getColName().toLowerCase();
         statsObj.setColName(colName);
+        statsObj.setColType(statsObj.getColType().toLowerCase());
         startFunction("write_partition_column_statistics:  db=" + dbName + " table=" + tableName +
           " part=" + partName + "column=" + colName);
       }
-- 
1.7.9.5

